x-pack/plugins/alerting/server/alerts_client/lib/build_recovered_alert.ts

```diff
@@ -95,6 +98,9 @@ export const buildRecoveredAlert = <
     [ALERT_FLAPPING]: legacyAlert.getFlapping(),
     // Set latest flapping_history
     [ALERT_FLAPPING_HISTORY]: legacyAlert.getFlappingHistory(),
+    // Alert is recovering from active state so by default it is improving
+    [ALERT_IS_IMPROVING]: true,
```

Comment:
by default if an alert is recovered, it is improving, regardless of whether severity level is defined.